hera_qm/xrfi.py

```diff
@@ -1983,34 +2068,40 @@ def day_threshold_run(data_files, history, nsig_f=7., nsig_t=7.,
     # Read in the metrics objects
     filled_metrics = []
     for ext in mexts:
-        if ext != 'data_metrics':
-            # Fill in 2nd metrics with 1st metrics where 2nd are not available.
+        # Fill in 2nd metrics with 1st metrics where 2nd are not available.
+        files1_all = [glob.glob(d + '/*' + ext + '1.h5') for d in xrfi_dirs]
+        files2_all = [glob.glob(d + '/*' + ext + '2.h5') for d in xrfi_dirs]
+        # only consider flagging products that exist in all observations for thresholding.
+        if np.all([len(f) > 0 for f in files1_all]) and np.all([len(f) > 0 for f in files2_all]):
             files1 = [glob.glob(d + '/*' + ext + '1.h5')[0] for d in xrfi_dirs]
             files2 = [glob.glob(d + '/*' + ext + '2.h5')[0] for d in xrfi_dirs]
             uvf1 = UVFlag(files1)
             uvf2 = UVFlag(files2)
             uvf2.metric_array = np.where(np.isinf(uvf2.metric_array), uvf1.metric_array,
                                          uvf2.metric_array)
             filled_metrics.append(uvf2)
-        else:
-            # Data was only run in second iteration
+        elif np.all([len(f) > 0 for f in files2_all]):
```

Comment:
Do we want to consider the possibility that some files will exist in round 1 but not round 2?

Reply:
I now consider this possibility as well.